File: src/C#/BehavioralPatterns/Mediator/MultipleExamples/CQRS/ProductsController.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace MultipleExamples.CQRS
{
    public class ProductsController
    {
        private readonly IMediator _mediator;

        public ProductsController(IMediator mediator)
        {
            _mediator = mediator;
        }

        public void CreateProduct(string productName, decimal productPrice)
        {
            _mediator.Send(new CreateProductCommand(productName, productPrice));
        }

        public object GetProducts()
        {
            return _mediator.Send(new GetProductsRequest());
        }
    }
}

File: src/PHP/BehavioralPatterns/Mediator/CQRS/ProductsController.php
<?php

namespace BehavioralPatterns\Mediator\CQRS;

class ProductsController {
    private IMediator $mediator;

    public function __construct($mediator) {
        $this->mediator = $mediator;
    }

    public function createProduct($productName, $productPrice): void
    {
        $this->mediator->send(new CreateProductCommand($productName, $productPrice));
    }

    public function getProducts() {
        return $this->mediator->send(new GetProductsRequest());
    }
}